### PR TITLE
Ioana AF/tema8

### PR DESCRIPTION
**Pull request by ioanaandroflorea**

### CODE REVIEW

File: src/test/java/ioanandroflorea/tests/Tema8/pageObjects/RefreshPage.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+        this.driver = driver;
+        wait = new WebDriverWait(driver, Duration.ofSeconds(5));
+        PageFactory.initElements(driver,this);
+    }
+    @FindBy(id ="refreshdate")
+    WebElement refreshButton;
+    public String getCurrentTimestamp(){
+        return refreshButton.getText();
+    }
+    public String getRefreshedTimestamp(){
```
Comment (95teo) [inline src/test/java/ioanandroflorea/tests/Tema8/pageObjects/RefreshPage.java:29]: This method does the same thing as getCurrentTimestamp(). No need to duplicate it. Keep only one of them and use it whenever you need it and remove the other one. Thanks!

File: src/test/java/ioanandroflorea/tests/Tema8/pageObjects/DisableDynamicButtonsPage [hunk 2]
```diff
@@ -0,0 +1,76 @@
+package ioanandroflorea.tests.Tema8.pageObjects;
```
Comment (95teo) [inline src/test/java/ioanandroflorea/tests/Tema8/pageObjects/DisableDynamicButtonsPage:1]: Please put it in a .java class file. I think you added it only as a simple file. Please also pay attention to the name of the class and file: DisabledDynamicButtonsPage vs DisableDynamicButtonsPage. They should be the same. Thanks!

**Review by 95teo — CHANGES_REQUESTED** [review]

File: src/test/java/ioanandroflorea/tests/Tema8/pageObjects/DisableDynamicButtonsPage (hunk 2 above)
Comment (ioanaandroflorea) [reply src/test/java/ioanandroflorea/tests/Tema8/pageObjects/DisableDynamicButtonsPage:1]: I realized I had an error here, but I didn't know where it came from. I hadn't looked carefully at the names of the classes.

**Review by redandreea — APPROVED** [review]

**Review by SanduBogdan95 — APPROVED** [review]

**Review by tincageorgianaalina — APPROVED** [review]

File: src/test/java/ioanandroflorea/tests/Tema8/teste/DisabledDynamicButtonsTests.java [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+        DisabledDynamicButtonsPage disabledDynamicButtonsPage = new DisabledDynamicButtonsPage(driver);
+        disabledDynamicButtonsPage.waitForTheStartButtonToBeClickable();
+        disabledDynamicButtonsPage.clickOnStartButton();
+        disabledDynamicButtonsPage.waitForOneButtonToBeClickable();
+        disabledDynamicButtonsPage.clickOnOneButton();
+        disabledDynamicButtonsPage.waitForTheMessageToDisappear();
+        disabledDynamicButtonsPage.clickOnSecondButton();
+        disabledDynamicButtonsPage.waitForTheMessageToDisappear();
+        disabledDynamicButtonsPage.clickOnThirdButton();
+        Assert.assertEquals(disabledDynamicButtonsPage.confirmationMessage(), "Click Buttons In Order", "All Buttons Clicked");
```
Comment (95teo) [inline src/test/java/ioanandroflorea/tests/Tema8/teste/DisabledDynamicButtonsTests.java:25]: The expected text should be "All Buttons Clicked" and not "Click Buttons In Order". Please make this change. Thanks!

File: src/test/java/ioanandroflorea/tests/Tema8/pageObjects/DisabledDynamicButtonsPage.java [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+    public void waitForTheStartButtonToBeClickable() {
+        wait.until(ExpectedConditions.visibilityOf(startButton));
+    }
+
+    public void clickOnStartButton() {
+        startButton.click();
+    }
+
+    public void waitForOneButtonToBeClickable() {
+        wait.until(ExpectedConditions.visibilityOfElementLocated(By.id("button01")));
```
Comment (95teo) [inline src/test/java/ioanandroflorea/tests/Tema8/pageObjects/DisabledDynamicButtonsPage.java:44]: Replace visibilityOfElementLocated with elementToBeClickable. The problem is with the expected condition you are using. The elements are already loaded in the page from the beginning so the are visible from the start. What you need here is to wait for them to be clickable as they are disabled until previous buttons are clicked.

File: src/test/java/ioanandroflorea/tests/Tema8/teste/BasicAjaxTests.java [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+import ioanandroflorea.tests.Tema8.pageObjects.BasicAjaxConfirmationPage;
+import ioanandroflorea.tests.Tema8.pageObjects.BasicAjaxPage;
+
+public class BasicAjaxTests extends BaseTest {
+
+    @Test(groups = {"mobile","desktop"})
+    public void codeItInTest() {
+        driver.get("https://testpages.herokuapp.com/styled/basic-ajax-test.html");
+        BasicAjaxPage basicAjaxPage = new BasicAjaxPage(driver);
+        basicAjaxPage.selectCategory("Desktop.xml");
```
Comment (95teo) [inline src/test/java/ioanandroflorea/tests/Tema8/teste/BasicAjaxTests.java:14]: From what I could saw, the category name is "Desktop" and not "Desktop.xml". This is causing the failure for this test. Please make the according changes in order to make the test to pass.

File: src/test/java/ioanandroflorea/tests/Tema8/teste/SimpleDynamicButtonsTests.java [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+import org.testng.Assert;
+import org.testng.annotations.Test;
+import ioanandroflorea.tests.Tema8.pageObjects.BasicAjaxConfirmationPage;
+import ioanandroflorea.tests.Tema8.pageObjects.BasicAjaxPage;
+import ioanandroflorea.tests.Tema8.pageObjects.SimpleDynamicButtonsPage;
+
+public class SimpleDynamicButtonsTests extends BaseTest {
+
+    @Test
+    public void verifyClickOnAllButtons() {
```
Comment (95teo) [inline src/test/java/ioanandroflorea/tests/Tema8/teste/SimpleDynamicButtonsTests.java:12]: You're missing groups for this test. Please add them. Thanks!

**Review by 95teo — CHANGES_REQUESTED** [review]

File: src/test/java/ioanandroflorea/tests/Tema8/teste/DisabledDynamicButtonsTests.java (hunk 3 above)
Comment (ioanaandroflorea) [reply src/test/java/ioanandroflorea/tests/Tema8/teste/DisabledDynamicButtonsTests.java:25]: I changed it, but the test still fails. I don't know what I should do.

File: src/test/java/ioanandroflorea/tests/Tema8/pageObjects/DisabledDynamicButtonsPage.java [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+import java.time.Duration;
+
+public class DisabledDynamicButtonsPage {
+
+    ChromeDriver driver;
+    WebDriverWait wait;
+
+    public DisabledDynamicButtonsPage(ChromeDriver driver) {
+        this.driver = driver;
+        wait = new WebDriverWait(driver, Duration.ofSeconds(5));
```
Comment (95teo) [inline src/test/java/ioanandroflorea/tests/Tema8/pageObjects/DisabledDynamicButtonsPage.java:19]: Try to increase the max time for wait. Try with a higher value than 5 and see which one make the test pass. Thanks!

**Review by 95teo — CHANGES_REQUESTED** [review]

**Review by 95teo — APPROVED** [review]

**Pull request merged**